engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/container/MultiInstanceBodyProcessor.java

```diff
@@ -219,18 +228,13 @@ public void onChildTerminated(
       final ExecutableMultiInstanceBody element,
       final BpmnElementContext flowScopeContext,
       final BpmnElementContext childContext) {
-    // triggered by completion condition of multi-instance
-    final Either<Failure, Boolean> failureBooleanEither =
-        canBeCompletedWithCompletionCondition(element, flowScopeContext);
-    if (failureBooleanEither.isRight() && failureBooleanEither.get()) {
-      if (stateBehavior.canBeCompleted(childContext)) {
-        stateTransitionBehavior.completeElement(flowScopeContext);
-      }
-    } else {
-      if (flowScopeContext.getIntent() == ProcessInstanceIntent.ELEMENT_TERMINATING
-          && stateBehavior.canBeTerminated(childContext)) {
-        terminate(element, flowScopeContext);
-      }
+    if (flowScopeContext.getIntent() == ProcessInstanceIntent.ELEMENT_TERMINATING
+        && stateBehavior.canBeTerminated(childContext)) {
+      terminate(element, flowScopeContext);
+    } else if (stateBehavior.canBeCompleted(childContext)) {
```

Comment:
🔧 Although this currently doesn't lead to wrong executions, the statement is not structured by what it semantically intents to achieve. Please split the if (a && b) into 2 nested ifs or add `flowScopeContext.getIntent() == ProcessInstanceIntent.ELEMENT_COMPLETING &&` to the else if statement.